### DISCUSSION TITLE
Raspberry Pi Zero 2

### DISCUSSION THREAD

**Discussion by weberbox** [nebhead/PiFire#50] [General]
Just saw this news today. Wonder if it would bring better performance to PiFire.  They say up to 5x faster.

https://www.raspberrypi.com/products/raspberry-pi-zero-2-w/

I just ordered another Zero to add a antenna to my build and have a bench test setup. Should have waited. Now I need to order this and try it 😀

**Comment by nebhead** [comment]
> Just saw this news today. Wonder if it would bring better performance to PiFire. They say up to 5x faster.

I saw that too!  Actually, I saw this YouTube video (https://youtu.be/lKS2ElWQizA) which made we aware of the other similar RPi Zero W alternatives as well.  Definitely could be a great replacement for the existing PiFire with better performance.  I wish they had included 5Ghz WiFi though - which would definitely have been a benefit over the original version.  

One thing that this video got me wondering is if I should be paying more attention to power consumption (especially when idle).  Maybe disable the HDMI port since it's not being used.

**Reply by weberbox** [reply]
Just watched the video. Very interesting.  I also watched this video https://youtu.be/Qxcyb5z2HDM

I want it just for the boot time. 3x faster. The boot now on the zero is a bit slow. I don't usually keep my grill plugged in so speeding up boot would be awesome. 

I am also interested in the pi alternatives. I am going to take a look. I like the huge community with the pi but getting a zero 2 might be tough due to shortage these days. I went to pre order one on a few sites and the shipping costs nearly the same as the pi 😞

**Reply by nebhead** [reply]
Oh wow!  Yes, boot time improvements would be greatly appreciated.  I actually have a few other projects that would benefit from faster boot.  I'm checking out the video now.  

Adafruit has them in stock and you can select First Class USPS shipping for around $5.50.  Not too shabby.  

https://www.adafruit.com/product/5291

Just picked one up. :)

**Reply by weberbox** [reply]
Nice!! I checked them earlier and it was out of stock. Just got one ordered can't wait to see what improvement it brings.

Now I need to order another ufl connector to solder on and figure out what other projects I want to do with some extra Zero 1's lol

**Reply by weberbox** [reply]
So I soldered up the GPIO and ufl connector today and put the zero 2 in the grill. 

It boots way faster, the web ui is way faster and it just loads a lot smoother. Only issue I am having is the spi display has lines through it. I am not sure if it needs a config changed or I messed up a wire somewhere. I had to leave shortly after installing so have not had time to troubleshoot yet. 

![20211106_232857](https://user-images.githubusercontent.com/313328/140637093-595162f0-60fd-4a5b-a361-3e5216c38c4b.jpg)

Have you played with yours at all yet?

**Comment by nebhead** [comment]
The PiZero2 shipping and arrived super fast (only three days!) so I do have
it, but I hadn't yet soldered it up yet.  I have had a few other projects
to work on this weekend and family stuff so I hadn't had a chance to get to
it yet.  It's possible the SPI has some different behavior or timing is
different with this version?  Did you upgrade the OS or are you using the
same SD Card?

**Reply by weberbox** [reply]
Ya I was super surprised how fast mine arrived it was like 3 days from New York. I expected a couple weeks lol. 

At first I used the same SD card from the zero w but then pulled it, wiped and started fresh. Same issue unfortunately.  I checked the wiring and all is good along with proper voltage so it is probably a timing issue of some sort. I know almost nothing about spi display stuff so googling is in my future haha.

**Reply by nebhead** [reply]
> Ya I was super surprised how fast mine arrived it was like 3 days from New York. I expected a couple weeks lol.

Same!  I'm always going to use USPS First Class in the future.  It had tracking and everything.  And it was a fraction of the price. 

> At first I used the same SD card from the zero w but then pulled it, wiped and started fresh. Same issue unfortunately. I checked the wiring and all is good along with proper voltage so it is probably a timing issue of some sort. I know almost nothing about spi display stuff so googling is in my future haha.

Oh man... There are bound to be a few bugs to work out with a new product so maybe this is one of those issues.  I hope we can still use the Luma Core.

**Reply by nebhead** [reply]
FYI, I just saw today that they posted the latest OS (Bullseye) to raspberrypi.org (https://downloads.raspberrypi.org/raspios_lite_armhf/release_notes.txt).  It's data 10-30-21, but I swear I looked over the weekend and the latest was still Buster from earlier this year.  Might be worth another shot... or it might have more bugs.  :P

**Reply by weberbox** [reply]
I checked over the weekend as well and it was Buster from May I think.

Anyway I am pretty sure I got the issue resolved and as a bonus the screen appears to be refreshing at a much higher rate. It looks a lot better when changing images or menu items. 

I cannot test with a regular zero w as it is a pain to swap pi's on the grill and I have not setup a bench test unit yet. If we can get this tested on a normal zero w and it does not break anything we can get it committed to dev branch if you want.

display_ili9341b.py
```
self.serial = spi(port=0, device=0, gpio_DC=24, gpio_RST=25, bus_speed_hz=32000000, reset_hold_time=0.2, reset_release_time=0.2)
self.device = ili9341(self.serial, active_low=False, width=self.WIDTH, height=self.HEIGHT, gpio_LIGHT=5)
```

Adding reset_hold_time and reset_release_time solved the lines timing issue and in the experimenting bus_speed_hz made a huge improvement in the display refresh rate. 

Some more testing would be good but I soft and hard rebooted the pi multiple times and the issues with the screen are gone. 

I have not run a full cook yet but did start the grill a few times and everything else seems to be working normally so far. I would recommend to wipe the SD and start fresh as I did have some strange issues with the distance sensor when I just swapped the sd from the zero to the zero 2. I just saved the pellet and settings json from the zero and then dropped them into the new install on the zero 2 overwriting the freshly created ones and rebooted. 

Also the ant built debs are working great for other and pi variants. Just run the build script and get the deb then sudo apt install ./xxx.deb and pifire is installed. Makes it easy to install updates etc.

**Comment by nebhead** [comment]
Awesome work!  I'm happy to give it a quick spin on my grill and let you
know.  Probably going to be tomorrow though.  It's starting to rain outside
and I'll wait until that passes.

**Reply by weberbox** [reply]
Sounds good whenever no rush. I still need to test more and make sure it stays issue free on the zero 2. I hope get the first cook with the zero 2 done this weekend but it is a busy weekend so not sure it will happen.

**Reply by nebhead** [reply]
Just tested the change and it works just fine!  And like you said, seems to be more responsive which is a huge bonus.  Tested through a reboot as well and looks to be fine so far.  I'd say we go ahead and commit to the dev branch.

**Comment by tukutt** [comment]
Hi,
It's very difficult to find a RPI Zero. Do you think alternative board like Banana Pi BPI-M2 Zero, Orange Pi Zero or MangoPi. Other recommandation ?
Thanks you !

PS : Maybe a simple RPI 3 ?!

**Reply by nebhead** [reply]
@tukutt - This project _should_ work any any of the Raspberry Pi models maybe with the exception of the very old Raspberry Pi's.  I haven't tried this on any of the alternative boards, but I am intrigued (particularly the Radxa Zero 2 board), and would like to pick one up some day and give it a shot.

**Reply by tukutt** [reply]
I have a RPI 4 in stock, i will install to test ;) Radxa Zero 2 with 2GB RAM and 8GB MMC 😍 I want try it too !
